src/biome/text/tokenizer.py

```diff
@@ -109,48 +99,50 @@ def _tokenize(self, text: str) -> List[Token]:
         tokens: `List[Token]`
 
         """
-        return self._base_tokenizer.tokenize(text[: self.max_sequence_length])
+        tokens = self.nlp(text[: self.max_sequence_length])
+        return self._sanitize(_remove_spaces(tokens) if self._remove_spaces else tokens)
 
     def tokenize_document(self, document: List[str]) -> List[List[Token]]:
         """ Tokenizes a document-like structure containing lists of text inputs
 
-        Use this to account for hierarchical text structures (e.g., a paragraph is a list of sentences)
+        Use this to account for hierarchical text structures (e.g., a paragraph)
 
         Parameters
         ----------
         document: `List[str]`
-            A `List` with text inputs, e.g., sentences
-            
+            A `List` with text inputs, e.g., paragraphs
+
         Returns
         -------
         tokens: `List[List[Token]]`
         """
-        sentences = [self.text_cleaning(text) for text in document]
+        paragraphs = [self.text_cleaning(text) for text in document]
         if self.segment_sentences:
-            sentences = [
-                sentence
-                for sentences in self.segment_sentences.batch_split_sentences(sentences)
-                for sentence in sentences
+            paragraphs = [
```

Comment:
The name can be a bit confusing, `paragraphs` becomes a list of sentences here, right?

Reply:
I think the problem here is the assumption that a document (list of text) is a list of sentences.

Document inputs can be generated by splitting a text by `\n`. In that case, the document may not be necessarily a list of sentences, but of paragraphs:

```python
doc = """
This is a sentence. This is another sentence.
One more sentence here
""".split("\n")

# In this case, doc will be:
doc == [ "This is a sentence. This is another sentence.", "One more sentence here" ] 

# ...and you still would like to split by sentence
[ "This is a sentence", "This is another sentence", "One more sentence here" ]
```

Reply:
If you want the hierarchy to be on sentence level, then you could just omit the first `.split("\n")` and pass in the whole `doc`.
Just saying that it is a bit counter intuitive that by activating `segment_sentences` you also through away the paragraph hierarchy.